compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -11486,7 +11487,8 @@ void
 OMR::Z::TreeEvaluator::forwardArrayCopySequenceGenerator(TR::Node *node, TR::CodeGenerator *cg, TR::Register *byteSrcReg, TR::Register *byteDstReg, TR::Register *byteLenReg, TR::Node *byteLenNode, TR_S390ScratchRegisterManager *srm, TR::LabelSymbol *mergeLabel)
    {
 #ifdef J9_PROJECT_SPECIFIC
-   bool mustGenerateOOLGuardedLoadPath = TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads() &&
+   bool mustGenerateOOLGuardedLoadPath = !TR::Compiler->om.shouldReplaceGuardedLoadWithSoftwareReadBarrier() &&
+                                         TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads() &&
```

Comment:
Why are the hardware/software paths different here?

Reply:
This is how the new software concurrent scavenge refArrayCopy differ from the old hardware concurrent scavenge arrayCopy. 
The old array copy uses LGG loop to copy arrays; whereas the new SW-CS simply calls a helper if it's in concurrent scavenge GC cycle. Hence, the difference between HW and SW paths.

Reply:
I'm not sure I'm following. Have we not decided that on the OpenJ9 side we will fall back to the reference arraycopy helper even for the HW-CS? I thought we determined it will likely be faster to just call the helper to do the array copy rather than potentially hitting the read barrier on every single loop iteration.

Reply:
Did we carry out this performance experiment?

Reply:
We haven't carried out the performance experiments. Without performance results, it's better to keep the HW-CS path as is until one is proven to be better than the other. Maybe I should evaluate these two and finalize the design now?

Reply:
Yes, I think we should do this now. No better time IMO.

Reply:
Performance evaluation on helper vs load-store-loop has been done. My current pull requests use load-store-loops when CS is active for both HW and SW CS.